glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -219,26 +246,59 @@ open class GleanInternalAPI internal constructor () {
      */
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun testIsExperimentActive(experimentId: String): Boolean {
-        Log.e(LOG_TAG, "testIsExperimentActive is a stub")
-        // TODO: 1552471 stub
-        // return ExperimentsStorageEngine.getSnapshot()[experimentId] != null
-        assert(false, { "testIsExperimentActive is a stub" })
-        return false
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        return LibGleanFFI.INSTANCE.glean_experiment_test_is_active(handle, experimentId).toBoolean()
+    }
+
+    /**
+     * Utility function to get a String -> String [Map] out of a [JSONObject].
+     */
+    private fun getMapFromJSONObject(jsonRes: JSONObject): Map<String, String>? {
+        return jsonRes.optJSONObject("extra")?.let {
+            val map = mutableMapOf<String, String>()
+            it.names()?.let { names ->
+                for (i in 0 until names.length()) {
+                    map[names.getString(i)] = it.getString(names.getString(i))
+                }
+            }
+            map
+        }
```

Comment:
I'm *shocked* this isn't somehow built-in to the Android JSON library, but I can't find it...

Reply:
Likewise! I spent some time looking for it and was tempted to add an extension for it :P